vars/buildPlugin.groovy

```diff
@@ -29,24 +29,39 @@ def call(Map params = [:]) {
   boolean archivedArtifacts = false
   Map tasks = [failFast: failFast]
   getConfigurations(params).each { config ->
-    String label = config.platform
+    String label = ''
+    String platform = config.platform
     String jdk = config.jdk
     String jenkinsVersion = config.jenkins
     if (config.containsKey('javaLevel')) {
       infra.publishDeprecationCheck('Remove javaLevel', 'Ignoring deprecated "javaLevel" parameter. This parameter should be removed from your "Jenkinsfile".')
     }
 
-    String stageIdentifier = "${label}-${jdk}${jenkinsVersion ? '-' + jenkinsVersion : ''}"
+    String stageIdentifier = "${platform}-${jdk}${jenkinsVersion ? '-' + jenkinsVersion : ''}"
     boolean first = tasks.size() == 1
     boolean skipTests = params?.tests?.skip
     boolean addToolEnv = !useContainerAgent
 
-    if (useContainerAgent && (label == 'linux' || label == 'windows')) {
-      def agentContainerLabel = jdk == '8' ? 'maven' : 'maven-' + jdk
-      if (label == 'windows') {
-        agentContainerLabel += '-windows'
+    if (useContainerAgent) {
+      if (platform == 'linux' || platform == 'windows') {
+        def agentContainerLabel = jdk == '8' ? 'maven' : 'maven-' + jdk
+        if (platform == 'windows') {
+          agentContainerLabel += '-windows'
+        }
+        label = agentContainerLabel
+      }
+    } else {
+      switch(platform) {
+        case 'windows':
+          label = 'docker-windows'
+          break
+        case 'linux':
+          label = 'vm && linux'
+          break
+        default:
+          echo "WARNING: Unknown platform '${platform}'. Agent label set to fallback value 'linux'"
```

Comment:
This is incorrect. The label should be left as is. For example https://github.com/jenkinsci/docker-workflow-plugin/runs/9836537148 ran the `maven-windows` branch from https://github.com/jenkinsci/docker-workflow-plugin/blob/d5d2e5c4007f7ea006152542b2bcbe0f1b2b08aa/Jenkinsfile#L3 on Linux (duplicating the designated Linux branch).

Reply:
https://github.com/jenkins-infra/pipeline-library/pull/658